packages/lesswrong/components/votes/UltraFeedReactionsPalette.tsx

```diff
@@ -0,0 +1,215 @@
+  const { reaction } = props;
+  return <div className={classes.tooltipRoot}>
+   <div className={classes.tooltipIcon}>
+      <ReactionIcon inverted={true} react={reaction.name} size={40}/>
+    </div>
+    <div>
+      <span className={classes.tooltipLabel}>{reaction.label}</span>
+      <ReactionDescription reaction={reaction} />
+    </div>
+  </div>
+}
+
+interface ListReactButtonProps {
+  getCurrentUserReactionVote: (name: EmojiReactName, quote: QuoteLocator|null) => VoteOnReactionType|null;
+  toggleReaction: (reactionName: string, quote: QuoteLocator|null) => void;
+  reaction: NamesAttachedReactionType;
+  placement?: PopperPlacementType;
+  size?: number;
+  quote: QuoteLocator|null;
+}
+
+const ListReactButton: React.FC<ListReactButtonProps> = ({
+  getCurrentUserReactionVote,
+  toggleReaction,
+  reaction,
+  placement = "left",
+  size = 18,
+  quote
+}) => {
+    const currentUserVote = getCurrentUserReactionVote(reaction.name, quote);
+    const classes = useStyles(styles);
+    return <LWTooltip
+      key={reaction.name} placement={placement}
+      title={<ReactTooltip reaction={reaction} />}
+    >
+      <div className={classes.paletteEntry}
+        onClick={_ev => toggleReaction(reaction.name, quote)}
+        key={reaction.name}
+      >
+        <span className={classNames(classes.paletteIcon, {
+          [classes.selected]: (currentUserVote==="created" || currentUserVote==="seconded"),
+          [classes.selectedAnti]: currentUserVote==="disagreed"})
+        }>
+          <ReactionIcon react={reaction.name} size={size}/>
+        </span>
+        <span className={classes.hoverBallotLabel}>
+          {reaction.label}
+        </span>
+      </div>
+    </LWTooltip>
+}
+
+const UltraFeedReactionsPalette = ({getCurrentUserReactionVote, toggleReaction, quote}: {
+  getCurrentUserReactionVote: (name: EmojiReactName, quote: QuoteLocator|null) => VoteOnReactionType|null,
+  toggleReaction: (reactionName: string, quote: QuoteLocator|null) => void,
+  quote: QuoteLocator|null,
+}) => {
+  const classes = useStyles(styles);
+  const { captureEvent } = useTracking()
+  const [searchText,setSearchText] = useState("");
+  const debouncedCaptureEvent = useRef(debounce(captureEvent, 500))
+  
+  const activeReacts = namesAttachedReactions.filter(r=>!r.deprecated);
+  const reactionsToShow = reactionsSearch(activeReacts, searchText);
+
+  // Ensure we always include the full set of active reactions in the same
+  // ordering as the master palette, to avoid divergence.
+  const orderedVisibleReactions = activeReacts.filter(r => reactionsToShow.includes(r));
+
+  return <div className={classes.moreReactions}>
+    <input
+      type="text" className={classes.searchBox}
+      value={searchText}
+      placeholder="Search reactions"
+      onChange={(ev) => {
+        setSearchText(ev.currentTarget.value)
+        debouncedCaptureEvent.current("ultraFeedReactPaletteSearch", {searchText: ev.currentTarget.value})
```

Comment:
I... sorta roll to disbelieve that we'll ever actually use this (vs. checking which reacts people actually leave)